elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -371,23 +455,73 @@ private static boolean validateTableConfig(ElideTableConfig elideTableConfig) {
             table.getDimensions().forEach(dim -> {
                 validateFieldNameUniqueness(tableFields, dim.getName(), table.getName());
                 validateSql(dim.getDefinition());
-                validateTableSource(elideTableConfig, dim.getTableSource());
+                validateTableSource(dim.getTableSource());
+                extractChecksFromExpr(dim.getReadAccess(), extractedChecks);
             });
 
             table.getMeasures().forEach(measure -> {
                 validateFieldNameUniqueness(tableFields, measure.getName(), table.getName());
                 validateSql(measure.getDefinition());
+                extractChecksFromExpr(measure.getReadAccess(), extractedChecks);
             });
 
             table.getJoins().forEach(join -> {
                 validateFieldNameUniqueness(tableFields, join.getName(), table.getName());
-                validateJoin(join, elideTableConfig);
+                validateJoin(join);
             });
+
+            extractChecksFromExpr(table.getReadAccess(), extractedChecks);
+            validateChecks(extractedChecks);
         }
 
         return true;
     }
 
+    private void validateChecks(Set<String> checks) {
+
+        if (checks.isEmpty()) {
+            return; // Nothing to validate
+        }
+
+        Set<String> staticChecks = dictionary.getCheckMappings().keySet();
+
+        List<String> undefinedChecks = checks
+                        .stream()
+                        .filter(check -> !(elideSecurityConfig.hasCheckDefined(check) || staticChecks.contains(check)))
+                        .collect(Collectors.toList());
+
+        if (!undefinedChecks.isEmpty()) {
+            throw new IllegalStateException("Found undefined security checks: " + undefinedChecks);
+        }
+    }
+
+    private static void extractChecksFromExpr(String readAccess, Set<String> extractedChecks) {
+
+        if (!isNullOrEmpty(readAccess)) {
+            ParseTree root = EntityPermissions.parseExpression(readAccess);
+            extractChecksFromTree(root, extractedChecks);
+        }
+    }
+
+    private static void extractChecksFromTree(ParseTree root, Set<String> extractedChecks) {
```

Comment:
Check out PermissionExpressionVisitor.

We should create a visitor to extract the check strings from the expression ParseTree rather than trying to manipulate the tree directly as we do here.